usr/lib/common/key_mgr.c

```diff
@@ -1483,41 +1490,102 @@ CK_RV key_mgr_derive_key(STDLL_TokData_t *tokdata,
         TRACE_ERROR("%s received bad argument(s)\n", __func__);
         return CKR_FUNCTION_FAILED;
     }
+
+    rc = object_mgr_find_in_map1(tokdata, base_key, &base_key_obj, READ_LOCK);
+    if (rc != CKR_OK) {
+        TRACE_ERROR("Failed to acquire key from specified handle.\n");
+        if (rc == CKR_OBJECT_HANDLE_INVALID)
+            rc = CKR_KEY_HANDLE_INVALID;
+        goto done;
+    }
+
+    rc = tokdata->policy->is_mech_allowed(tokdata->policy, mech,
+                                          &base_key_obj->strength,
+                                          POLICY_CHECK_DERIVE,
+                                          base_key_obj->session);
+    if (rc != CKR_OK) {
+        TRACE_ERROR("POLICY VIOLATION: derive key\n");
+        goto done;
+    }
+
+    if (!key_object_is_mechanism_allowed(base_key_obj->template,
+                                         mech->mechanism)) {
+        TRACE_ERROR("Mechanism not allwed per CKA_ALLOWED_MECHANISMS.\n");
+        rc = CKR_MECHANISM_INVALID;
```

Comment:
allowed

Reply:
fixed. 

Same typo at several other places, fixed those in an additional commit :-)